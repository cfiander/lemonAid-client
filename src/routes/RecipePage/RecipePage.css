.recipeHeader, .recipeList {
    text-align: center;
}


.recipeHeader, .recipeList {
    text-align: center;
}

.recipeList {
    list-style: none;
    font-family: 'Playfair Display', serif;
    text-shadow: 0px 4px 3px rgba(0,0,0,0.4),
             0px 8px 13px rgba(0,0,0,0.1),
             0px 18px 23px rgba(0,0,0,0.1);
}


.recipeSection{
    padding-top: 10%;
}

.recipeTitle {
  font-family: 'Playfair Display', serif;
  font-size: 250%;
  text-align: center;
  font-weight: bold;
  color:white;
  text-shadow: 0px 4px 3px rgba(0,0,0,0.4),
             0px 8px 13px rgba(0,0,0,0.1),
             0px 18px 23px rgba(0,0,0,0.1);
}

.expandButton {
    background-color: transparent;
    color: white;
    text-shadow: 0px 4px 3px rgba(0,0,0,0.4),
             0px 8px 13px rgba(0,0,0,0.1),
             0px 18px 23px rgba(0,0,0,0.1);
}

.expandButton, .collapseButton{
    height:10px;
    color: white;
}


