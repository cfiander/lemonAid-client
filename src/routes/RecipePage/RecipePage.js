import React, { Component } from 'react'
import RecipeContext from '../../contexts/RecipeContext'
import RecipeApiService from '../../services/recipes-api-service'
import Recipes from '../../components/Recipes/Recipes'
import './RecipePage.css'

export default class RecipePage extends Component {
    static contextType = RecipeContext

    componentDidMount() {
        
        setTimeout( () => {
            const formatted = RecipeApiService.formatIngredients(this.context.ingredientsList)
            RecipeApiService.getRecipes(formatted)
            .then(this.context.setRecipeList)
            .catch(this.context.setError)
        },600)
    }

    renderRecipes() {
        const {recipeList = []} = this.context
        return recipeList.map(recipe => 
            <Recipes 
                key={recipe.id}
                recipe={recipe}
            />
        )
    }

    render() {
        const {error} = this.context
        return (
                <div className="recipeSection">
                <h1 className="recipeTitle">Your Recipes</h1>
                {/* <select>
                    <option value="" selected disabled hidden>Rank Results</option>
                    <option value="0">Minimize Missing Ingredients</option>
                    <option value="1">Maximize Used Ingredients</option>
                    <option value="2">Relevance To Search</option>
                </select> */}
                <ul className="recipeList">
                {error ? <p className='red'>There was an error, try again</p>
                : this.renderRecipes()}
                </ul>
                </div>
                
        )
    }
}