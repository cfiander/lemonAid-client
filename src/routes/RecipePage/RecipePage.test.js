
import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import RecipePage from './RecipePage';

describe(`IngredientsPage component`, () => {
  const props = {
    recipeList: [
      {
        "id":645152,
        "title":"Grandma's Apple Crisp",
        "image":"https://spoonacular.com/recipeImages/Grandmas-Apple-Crisp-645152.jpg",
        "usedIngredientCount":3,
        "missedIngredientCount":6,
        "likes":1
      },
      {
        "id":645152,
        "title":"Grandma's Apple Crisp",
        "image":"https://spoonacular.com/recipeImages/Grandmas-Apple-Crisp-645152.jpg",
        "usedIngredientCount":3,
        "missedIngredientCount":6,
        "likes":1
      },
      {
    "id":645152,
    "title":"Grandma's Apple Crisp",
    "image":"https://spoonacular.com/recipeImages/Grandmas-Apple-Crisp-645152.jpg",
    "usedIngredientCount":3,
    "missedIngredientCount":6,
    "likes":1
      },
    ]
  }

  it('renders an RecipePage by default', () => {
    const wrapper = shallow(<RecipePage/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a recipe in ul for each recipe in the array', () => {
    const ul = shallow(<RecipePage {...props} />)
      .find('recipeList')
    expect(toJson(ul)).toMatchSnapshot()
  })
})