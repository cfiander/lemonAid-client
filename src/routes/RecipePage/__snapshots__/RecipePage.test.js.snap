// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngredientsPage component renders a Ingredient in ul for each Ingredient in array 1`] = `null`;

exports[`IngredientsPage component renders an RecipePage by default 1`] = `
<div
  className="recipeSection"
>
  <h1
    className="recipeTitle"
  >
    Your Recipes
  </h1>
  <ul
    className="recipeList"
  />
</div>
`;
